docs/concepts/CLQL.md

```diff
@@ -376,6 +376,35 @@ common.func:
 
 Note that CLQL elements that are children of `path`, not just the `if_stmt`. Also note that repeated definitions of `varName` are replaced with assertions.
 
+## Pathend
+
+Suppose we wanted to match triply nested if statements with a function call inside the innermost if statement. Without paths our query looks like:
+
+```clql
+common.if_stmt:
+  common.if_stmt:
+    common.if_stmt:
+      common.func_call
+```
+
+with paths our query looks like:
+
+```clql
+path(repeat = 3):
+  common.if_stmt:
+    pathcontinue
+  pathend:
+    common.func_call
+```
+
+## Caveats
+
+Branching, where `path` statement has multiple `pathcontinue` statements is currently not supported.
+
+Nested paths are not supported.
+
+Using `any_of` inside a path statement is not supported.
```

Comment:
Yes it will be supported in the future, that's a key use case.

Can you give an example of where you would be choosing between putting a `depth = any` inside the path or the path's children?

The path element only takes a `repeat` argument, facts inside path can take `depth` arguments. Element, by the way, is the generic term for fact, property, path, any_of, etc.

Yes you can have a path at the root.

Reply:
The case I had in mind when asking the question was the tenet I linked to above. Here I have "depth = any" specified for both children. But if a repeat was specified in the path then it each nested child would have a "depth = any". I'm unsure what this would do to performance, but I'd imagine you might get a tetration thing going. Also, it kind of makes sense to place the "depth = any" within the path fact because then you're only specifying it once. Either that or place the "depth = any" on a zero-width parent to the path fact. Should this go into discuss?

Reply:
I still don't know what you're suggesting. I can't think of any argument repeat argument that you could pass to path that would replace the need for a `depth = any`. Can you write up some CLQL?

Reply:
I made a topic on discuss to continue the conversation.
https://discuss.codelingo.io/t/clql-syntax-new-features/87